homeassistant/components/mqtt/client.py

```diff
@@ -806,7 +807,7 @@ def _mqtt_on_message(
         self, _mqttc: mqtt.Client, _userdata: None, msg: mqtt.MQTTMessage
     ) -> None:
         """Message received callback."""
-        self.hass.add_job(self._mqtt_handle_message, msg)
+        self.loop.call_soon_threadsafe(self._mqtt_handle_message, msg)
```

Comment:
Please always add a comment when using the loop directly instead of our core APIs so we know why that is done.

Reply:
See #108942 for a follow up